storage_test.go

```diff
@@ -0,0 +1,442 @@
+			},
+			wantErr: false,
+			expected: &Storage{
+				Table:               "Testing123",
+				AwsSession:          defaultAwsSession,
+				LockTimeout:         lockTimeoutMinutes,
+				LockPollingInterval: lockPollingInterval,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := &Storage{
+				Table:         tt.fields.Table,
+				AwsSession:    tt.fields.AwsSession,
+				AwsEndpoint:   tt.fields.AwsEndpoint,
+				AwsRegion:     tt.fields.AwsRegion,
+				AwsDisableSSL: tt.fields.AwsDisableSSL,
+			}
+			if err := s.initConfig(); (err != nil) != tt.wantErr {
+				t.Errorf("initConfig() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			// unset AwsSession since it is too complicated for reflection testing
+			s.AwsSession = tt.expected.AwsSession
+			if !reflect.DeepEqual(tt.expected, s) {
+				t.Errorf("Expected does not match actual: %+v != %+v. \nAwsSession \n\texpected: %+v, \n\tactual: %+v",
+					tt.expected, s, tt.expected.AwsSession, s.AwsSession)
+			}
+		})
+	}
+}
+
+func TestDynamoDBStorage_Store(t *testing.T) {
+	err := initDb()
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	type fields struct {
+		Table         string
+		KeyPrefix     string
+		ColumnName    string
+		AwsSession    *session.Session
+		AwsEndpoint   string
+		AwsRegion     string
+		AwsDisableSSL bool
+	}
+	type args struct {
+		key   string
+		value []byte
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "simple key/value store",
+			fields: fields{
+				Table:         TestTableName,
+				AwsEndpoint:   os.Getenv("AWS_ENDPOINT"),
+				AwsRegion:     os.Getenv("AWS_DEFAULT_REGION"),
+				AwsDisableSSL: DisableSSL,
+			},
+			args: args{
+				key:   "simple-key",
+				value: []byte("value"),
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := &Storage{
+				Table:         tt.fields.Table,
+				AwsSession:    tt.fields.AwsSession,
+				AwsEndpoint:   tt.fields.AwsEndpoint,
+				AwsRegion:     tt.fields.AwsRegion,
+				AwsDisableSSL: tt.fields.AwsDisableSSL,
+			}
+			if err := s.Store(tt.args.key, tt.args.value); (err != nil) != tt.wantErr {
+				t.Errorf("Store() error = %v, wantErr %v", err, tt.wantErr)
+			}
+			loaded, err := s.Load(tt.args.key)
+			if err != nil {
+				t.Errorf("failed to load after store: %s", err.Error())
+			}
+			if string(loaded) != string(tt.args.value) {
+				t.Errorf("Load() returned value other than expected. Expected: %s, Actual: %s", string(tt.args.value), string(loaded))
+			}
+		})
+	}
+}
+
+func TestDynamoDBStorage_List(t *testing.T) {
+	err := initDb()
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	storage := Storage{
+		Table:         TestTableName,
+		AwsEndpoint:   os.Getenv("AWS_ENDPOINT"),
+		AwsRegion:     os.Getenv("AWS_DEFAULT_REGION"),
+		AwsDisableSSL: DisableSSL,
+	}
+
+	prefix := "domain"
+
+	fixturesWithPrefix := map[string]string{
```

Comment:
It might be good to test with enough data to trigger the pagination code (i.e. more than one page of data).

In that case, instead of hardcoding this map, you can dynamically create it and the Store params inside the `for` loop.